tasks/firewall.yml

```diff
@@ -1,70 +1,18 @@
-  service_facts:
-
-- name: Configure firewalld
-  block:
-    - name: Enable service 'high-availability' in firewalld
-      command: firewall-cmd --add-service high-availability
-      when: ansible_facts.services['firewalld.service'].state == 'running'
-      register: __ha_cluster_firewall_cmd
-      failed_when:
-        - __ha_cluster_firewall_cmd.rc != 0
-        - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-    - name: Enable service 'high-availability' in firewalld permanent config
-      command: firewall-cmd --permanent --add-service high-availability
-      register: __ha_cluster_firewall_cmd
-      when: ansible_facts.services['firewalld.service'].state == 'running'
-      failed_when:
-        - __ha_cluster_firewall_cmd.rc != 0
-        - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-    - name: Enable service 'high-availability' in firewalld offline config
-      command: firewall-offline-cmd --add-service high-availability
-      register: __ha_cluster_firewall_cmd
-      when: ansible_facts.services['firewalld.service'].state != 'running'
-      failed_when:
-        - __ha_cluster_firewall_cmd.rc != 0
-        - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-    - name: Enable fence-virt port in firewalld - all options
-      when:
-        - ansible_architecture == 'x86_64'
-        - (
-            'fence-virt' in ha_cluster_fence_agent_packages
-            or
-            'fence-virt' in ha_cluster_extra_packages
-            or
-            'fence-agents-all' in ha_cluster_fence_agent_packages
-            or
-            'fence-agents-all' in ha_cluster_extra_packages
-          )
-      block:
-        - name: Enable fence-virt port in firewalld
-          command: firewall-cmd --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state == 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-        - name: Enable fence-virt port in firewalld permanent config
-          command: firewall-cmd --permanent --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state == 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
-        - name: Enable fence-virt port in firewalld offline config
-          command: firewall-offline-cmd --add-port 1229/tcp
-          when: ansible_facts.services['firewalld.service'].state != 'running'
-          register: __ha_cluster_firewall_cmd
-          failed_when:
-            - __ha_cluster_firewall_cmd.rc != 0
-            - __ha_cluster_firewall_cmd.rc != 11  # already enabled
-
+- name: Ensure the service and the ports status with the firewall role
+  include_role:
+    name: fedora.linux_system_roles.firewall
```

Comment:
This causes the role to fail on my VMs as the firewall role cannot be found. It seems like ansible collections need to be installed for this to work. Is the `rhel-system-roles` packages going to depend on ansible collections? Should this be `linux-system-roles.firewall` instead, so that it uses the firewall role from the `rhel-system-roles` package?

Reply:
> This causes the role to fail on my VMs

How are you running your VMs?  The `tox-lsr` tox plugin using `-e qemu[-ansible-core-2.x]` supports installing collection dependencies for the test.
We are following https://linux-system-roles.github.io/documentation/role-requirements.html
And if you use https://github.com/linux-system-roles/tox-lsr#qemu-testing e.g.
`tox -e qemu-ansible-core-2.13 -- --image-name centos-9 tests/tests_cluster_basic_new_psks.yml`
then tox will automatically install the dependencies from `meta/collection-requirements.yml` in your test environment (i.e. not in ~/.ansible/collections and "pollute" your local workstation).

> as the firewall role cannot be found. It seems like ansible collections need to be installed for this to work.

Yes.

> Is the `rhel-system-roles` packages going to depend on ansible collections?

For upstream (users installing the ha_cluster role from Galaxy), users will need to follow directions like this: https://github.com/linux-system-roles/storage/#requirements

@nhosoi The ha_cluster README needs to be updated with a `Requirements` section

Users using the ha_cluster role as part of the fedora.linux_system_roles collection will not need to do this, nor users using the role from the Fedora RPM package.

For downstream, the dependencies will be automatically handled, so the user will need to do nothing.

> Should this be `linux-system-roles.firewall` instead, so that it uses the firewall role from the `rhel-system-roles` package?

No, see above.

Reply:
Ok, thanks for explaining. I don't use tox. I'm actually developing and testing the role on VMs which I use for developing pcs as well. Once I installed the collection on the VMs, everything started working just fine.

Reply:
Thank you for the confirmation, @tomjelinek!
I'm updating the README.md with the new requirements and the examples as suggested by @richm and @spetrosi. (stay tuned...)

Reply:
@tomjelinek, @richm, @spetrosi, could you please take a look at https://github.com/linux-system-roles/ha_cluster/pull/69?
Thanks!